src/WinRT.Runtime/ExceptionHelpers.cs

```diff
@@ -51,133 +51,139 @@ static unsafe class ExceptionHelpers
-            string restrictedErrorReference = null;
-            string restrictedCapabilitySid = null;
-            bool hasOtherLanguageException = false;
-            string errorMessage = string.Empty;
-
-            if (useGlobalErrorState)
-            {
-                using var restrictedErrorInfoRef = BorrowRestrictedErrorInfo();
-                if (restrictedErrorInfoRef != null)
-                {
-                    restrictedErrorInfoToSave = restrictedErrorInfoRef.As<ABI.WinRT.Interop.IRestrictedErrorInfo.Vftbl>();
-                    ABI.WinRT.Interop.IRestrictedErrorInfo restrictedErrorInfo = new ABI.WinRT.Interop.IRestrictedErrorInfo(restrictedErrorInfoRef);
-                    restrictedErrorInfo.GetErrorDetails(out description, out int hrLocal, out restrictedError, out restrictedCapabilitySid);
-                    restrictedErrorReference = restrictedErrorInfo.GetReference();
-                    if (restrictedErrorInfoRef.TryAs<ABI.WinRT.Interop.ILanguageExceptionErrorInfo.Vftbl>(out var languageErrorInfoRef) >= 0)
-                    {
-                        ILanguageExceptionErrorInfo languageErrorInfo = new ABI.WinRT.Interop.ILanguageExceptionErrorInfo(languageErrorInfoRef);
-                        using IObjectReference languageException = languageErrorInfo.GetLanguageException();
-                        if (languageException is object)
-                        {
-                            if (languageException.IsReferenceToManagedObject)
-                            {
-                                ex = ComWrappersSupport.FindObject<Exception>(languageException.ThisPtr);
-                                if (GetHRForException(ex) == hr)
-                                {
-                                    restoredExceptionFromGlobalState = true;
-                                    return ex;
-                                }
-                            }
-                            else
-                            {
-                                // This could also be a proxy to a managed exception.
-                                hasOtherLanguageException = true;
-                            }
-                        }
-                    }
-
-                    if (hr == hrLocal)
+        [DllImport("oleaut32.dll")]
+        private static extern int SetErrorInfo(uint dwReserved, IntPtr perrinfo);
+
+        private static delegate* unmanaged[Stdcall]<IntPtr*, int> getRestrictedErrorInfo;
+        private static delegate* unmanaged[Stdcall]<IntPtr, int> setRestrictedErrorInfo;
+        private static delegate* unmanaged[Stdcall]<int, IntPtr, IntPtr, int> roOriginateLanguageException;
+        private static delegate* unmanaged[Stdcall]<IntPtr, int> roReportUnhandledError;
+
+        private static readonly bool initialized =  Initialize();
+
+        private static bool Initialize()
+        {
+            fixed (char* moduleName = "api-ms-win-core-winrt-error-l1-1-1.dll")

```

Comment:
This is making the code quite messy. I'm thinking we should consider doing what `[LibraryImport]` is doing, just manuall — keep managed signatures for the platform invoke APIs, and just implement them with blittable P/Invoke internally. Same performance and AOT-friendliness improvements, and much less error prone and clunky code elsewhere 🙂

Reply:
I've created an abstraction for `LoadLibraryExW`. As for the other methods, I don't think it affects code readability much. Will probably still create abstractions for them in case the user requires access to those methods (even if I'm not sure why).

Reply:
I've added the rest of the wrappers (except for `GetProcAddress`, which is being discussed in the other thread). Will flag this as resolved.